### PR TITLE
Format Time correctly with Formatter

### LINKED ISSUE #33: Support Formatter from ocean.time.Time
Right now `format("{}", someTimestamp)` prints `{ ticks_: 636247008000000000 }` instead of `03/10/17 00:00:00`

Good old Tango supports it via [this horror](https://github.com/sociomantic-tsunami/ocean/blob/286a81bbabb0f422c268c53642f97fd832648390/src/ocean/text/convert/Layout_tango.d#L859-L871).

With the `Formatter`, we can just provide a `toString` method. Feel free to assign me.

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
CC @andrej-mitrovic-sociomantic : Do you need other format string than 'G' ?

Fixes #33

### CODE REVIEW

File: src/ocean/text/convert/DateTime_tango.d [hunk 1]
```diff
@@ -586,11 +604,9 @@ struct DateTimeLocale
         return f;
     }
 
-    /**********************************************************************
-
-     **********************************************************************/
-
-    private char[] formatCustom (ref Result result, Time dateTime, cstring format)
+    /// Implementation of format
```
Comment (gavin-norman-sociomantic) [inline src/ocean/text/convert/DateTime_tango.d]: Would be nice to add doc in the standard format, if you're going to add it. (Despite this module's doc being very non-standard ;)

File: src/ocean/text/convert/DateTime_tango.d [hunk 2]
```diff
@@ -586,11 +604,9 @@ struct DateTimeLocale
...
+    private void formatCustom (size_t delegate(cstring) sink, Time dateTime,
```
Comment (gavin-norman-sociomantic) [inline src/ocean/text/convert/DateTime_tango.d]: Suggest splitting this change to `formatCustom` to accept a sink delegate into a separate commit. It's not clear to me where the "Format Time correctly with Formatter" part comes in.

File: src/ocean/text/convert/DateTime_tango.d (hunk 1 above)
Comment (mathias-lang-sociomantic) [reply src/ocean/text/convert/DateTime_tango.d]: I'd rather keep any cleanup for another day (I prefer to do all at once). Will revert :)

**Comment by andrej-mitrovic-sociomantic** [comment]
> Do you need other format string than 'G' ?

I only need whatever dolphin's output needs. I don't recall which format string it is though.

**Comment by andrej-mitrovic-sociomantic** [comment]
I'm not sure what `<redacted>` is using. But if the tests of `<redacted>` pass, we're good to go, stranger.

**Comment by mathias-lang-sociomantic** [comment]
Updated:
- Reverted the minor doc change
- Split into 2 commits.

**Comment by mathias-lang-sociomantic** [comment]
Also rebased on v2.5.x

**Comment by andrej-mitrovic-sociomantic** [comment]
You have a typo in the commit message: `speficier`

**Comment by mathias-lang-sociomantic** [comment]
Made the message less `speficic`

**Comment by mathias-lang-sociomantic** [comment]
Ping

**Comment by mihails-strasuns-sociomantic** [comment]
@andrej-mitrovic-sociomantic can you please confirm that this works for you?

**Comment by andrej-mitrovic-sociomantic** [comment]
@mathias-baumann-sociomantic can you please confirm that it works for me?

**Comment by andrej-mitrovic-sociomantic** [comment]
wrong mathias

**Comment by andrej-mitrovic-sociomantic** [comment]
@mathias-lang-sociomantic ^

**Comment by mathias-lang-sociomantic** [comment]
It does

**Comment by mihails-strasuns-sociomantic** [comment]
Admirable teamwork :)

**Comment by mihails-strasuns-sociomantic** [comment]
@mathias-lang-sociomantic if this goes as a feature, should also do release notes mention

**Comment by mihails-strasuns-sociomantic** [comment]
Marking as a bug fix after chat with @mathias-lang-sociomantic

**Pull request merged**